service/matching/taskListManager.go

```diff
@@ -293,6 +303,25 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 	scope := metrics.MatchingTaskListMgrScope
 	timer := time.NewTimer(c.config.LongPollExpirationInterval)
 	defer timer.Stop()
+
+	pollerID, ok := ctx.Value(pollerIDKey).(string)
+	childCtx, cancel := context.WithCancel(ctx)
+	if ok {
+		// Found pollerID on context, add it to the map to allow it to be canceled in
+		// response to CancelPoller call
+		c.outstandingPollsLock.Lock()
+		c.outstandingPollsMap[pollerID] = cancel
+		c.outstandingPollsLock.Unlock()
+		defer func() {
+			c.outstandingPollsLock.Lock()
+			delete(c.outstandingPollsMap, pollerID)
+			c.outstandingPollsLock.Unlock()
+			cancel()
+		}()
+	} else {
```

Comment:
do we expect this to happen?

Reply:
This is just for backwards compatibility.